### PR TITLE
Refactor amm methods

### PR DESCRIPTION
**Pull request by jepidoptera**
all tests passing

### CODE REVIEW

File: hydradx/model/amm/omnipool_amm.py [hunk 1]
```diff
@@ -297,795 +1016,64 @@ def __init__(self, state: OmnipoolState):
...
-def swap_lrna(
+def simulate_swap_lrna(
         old_state: OmnipoolState,
         old_agent: Agent,
         delta_ra: float = 0,
         delta_qa: float = 0,
         tkn: str = '',
         modify_imbalance: bool = True
 ) -> tuple[OmnipoolState, Agent]:
-    """Compute new state after LRNA swap"""
```
Comment (poliwop) [inline hydradx/model/amm/omnipool_amm.py:1080]: This probably should stay "state"?

File: hydradx/model/amm/omnipool_amm.py [hunk 2]
```diff
@@ -1099,65 +1087,67 @@ def swap(
...
 
-OmnipoolState.swap = staticmethod(swap)
-OmnipoolState.execute_swap = staticmethod(execute_swap)
-OmnipoolState.add_liquidity = staticmethod(add_liquidity)
-OmnipoolState.execute_add_liquidity = staticmethod(execute_add_liquidity)
-OmnipoolState.remove_liquidity = staticmethod(remove_liquidity)
-OmnipoolState.usd_price = staticmethod(usd_price)
-OmnipoolState.lrna_price = staticmethod(lrna_price)
-OmnipoolState.price = staticmethod(price)
+# OmnipoolState.swap = staticmethod(simulate_swap)
```
Comment (poliwop) [inline hydradx/model/amm/omnipool_amm.py:1137]: Delete these instead of commenting out?

File: hydradx/model/amm/stableswap_amm.py [hunk 3]
```diff
@@ -167,247 +167,255 @@ def __repr__(self):
...
+            new_agent.holdings[tkn_buy] = 0
+        new_agent.holdings[tkn_buy] += buy_quantity
+        new_agent.holdings[tkn_sell] -= sell_quantity
+        self.liquidity[tkn_buy] -= buy_quantity
+        self.liquidity[tkn_sell] += sell_quantity
+    
+        return self, new_agent
+    
+    #
+    # def execute_remove_liquidity_old(
```
Comment (poliwop) [inline hydradx/model/amm/stableswap_amm.py:201]: I think we can delete this

File: hydradx/tests/test_omnipool_amm.py [hunk 4]
```diff
@@ -448,9 +447,11 @@ def test_lrna_swap_buy_with_lrna_mint(
         feeless_state.last_fee[asset] = 0
 
     # Test with trader buying asset i
-    swap_state, swap_agent = oamm.swap_lrna(initial_state, old_agent, delta_ra, 0, i)
-    feeless_swap_state, feeless_swap_agent = oamm.swap_lrna(feeless_state, old_agent, delta_ra_feeless, 0, i)
-    feeless_spot_price = feeless_swap_state.price(feeless_swap_state, i)
+    swap_state = initial_state.copy().lrna_swap(old_agent.copy(), delta_ra, 0, i)
+    feeless_swap_state = feeless_state.copy().lrna_swap(old_agent.copy(), delta_ra_feeless, 0, i)
+    # swap_state, swap_agent = oamm.swap_lrna(initial_state, old_agent, delta_ra, 0, i)
```
Comment (poliwop) [inline hydradx/tests/test_omnipool_amm.py:452]: Delete commented out lines?

**Review by poliwop — CHANGES_REQUESTED** [review]
Looks good, very small changes, mostly deleting commented out code and stuff like that.

File: hydradx/model/run.py [hunk 5]
```diff
@@ -23,11 +23,11 @@ def run(initial_state: GlobalState, time_steps: int, silent: bool = False) -> li
         new_global_state.evolve()
 
         # agent actions
-        agent_ids = deepcopy(list(new_global_state.agents.keys()))
+        agent_ids = list(new_global_state.agents.keys())
```
[the remark this answers is not in the document]
Comment (jepidoptera) [reply hydradx/model/run.py:26]: Yes, the list() function itself already creates a new object.

File: hydradx/model/amm/stableswap_amm.py (hunk 3 above)
Comment (jepidoptera) [reply hydradx/model/amm/stableswap_amm.py:201]: I think so too, we're not going back

File: hydradx/model/processing.py [hunk 6]
```diff
@@ -52,13 +52,13 @@ def postprocessing(events: list, optional_params: list[str] = ()) -> list:
     if unrecognized_params:
         raise ValueError(f'Unrecognized parameter {unrecognized_params}')
 
-    # a little pre-processing
-    if 'deposit_val' in optional_params:
```
[the remark this answers is not in the document]
Comment (jepidoptera) [reply hydradx/model/processing.py:56]: I don't think we use any of this. I've moved into using pu.get_datastream() or just doing this stuff ad-hoc in the notebooks, if it's too case-specific to be standardized in a way that's worth maintaining.

File: hydradx/model/amm/omnipool_amm.py (hunk 1 above)
Comment (jepidoptera) [reply hydradx/model/amm/omnipool_amm.py:1080]: Yup. Lol

File: hydradx/model/amm/global_state.py [hunk 7]
```diff
@@ -205,6 +204,41 @@ def deposit_val(self, agent: Agent) -> float:
...
+    def external_market_trade(
+            self,
+            agent_id: str,
+            tkn_buy: str,
+            tkn_sell: str,
+            buy_quantity: float = 0,
+            sell_quantity: float = 0
+    ):
+        # do a trade at spot price on the external market
+        # this should maybe only work in USD, because we're probably talking about coinbase or something
```
[the remark this answers is not in the document]
Comment (jepidoptera) [reply hydradx/model/amm/global_state.py:216]: Not really related to this PR, but that does make sense. I can add it

**Review by poliwop — APPROVED** [review]
Looks good.

**Pull request merged**